spec/javascripts/health-care/components/social-security-number.spec.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import ReactTestUtils from 'react-addons-test-utils';
+import SocialSecurityNumber from '../../../../_health-care/_js/_components/social-security-number';
+import _ from 'lodash';
+
+describe('<SocialSecurityNumber>', () => {
+  let component = null;
+
+  beforeEach(() => {
+    component = ReactTestUtils.renderIntoDocument(
+      <SocialSecurityNumber ssn={{ssn:"999-99-9999"}} onUserInput={(update) => {}}/>
+    );
+    assert.ok(component, "Cannot even render component");
+  });
+
+  it('has sane looking features', () => {
+    const inputs = ReactTestUtils.scryRenderedDOMComponentsWithTag(component, 'input');
+    expect(inputs).to.have.length(1);
+  });
+
+  it('sets and removes error css on invalid SSN', () => {
+    // Initial state should be valid.
+    expect(component.state.hasError).to.be.false;
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithClass(
+        component, 'usa-input-error')).to.have.length(0);
+
+    // Valid SSN has no error.
+    component.refs.ssn.value = '123-45-6789';
+    ReactTestUtils.Simulate.change(component.refs.ssn);
+    expect(component.state.hasError).to.be.false;
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithClass(
+        component, 'usa-input-error')).to.have.length(0);
+
+    // SSN with alphabetical characters is invalid.
+    component.refs.ssn.value = '123-45-678a';
+    ReactTestUtils.Simulate.change(component.refs.ssn);
+    expect(component.state.hasError).to.be.true;
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithClass(
+        component, 'usa-input-error')).to.have.length(1);
+
+    // SSN without dashes is invalid.
+    component.refs.ssn.value = '123456789';
+    ReactTestUtils.Simulate.change(component.refs.ssn);
+    expect(component.state.hasError).to.be.true;
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithClass(
+        component, 'usa-input-error')).to.have.length(1);
+
+    // SSN with too few numbers is invalid.
+    component.refs.ssn.value = '123-45-678';
+    ReactTestUtils.Simulate.change(component.refs.ssn);
+    expect(component.state.hasError).to.be.true;
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithClass(
+        component, 'usa-input-error')).to.have.length(1);
+
+    // SSN with non-numberic characters is invalid.
+    component.refs.ssn.value = '#12-34-5678';
+    ReactTestUtils.Simulate.change(component.refs.ssn);
+    expect(component.state.hasError).to.be.true;
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithClass(
+        component, 'usa-input-error')).to.have.length(1);
+  });
+
+  it('includes ErrorMessage component when invalid SSN', () => {
```

Comment:
I wanted to write a test that actually checked if the component `ErrorMessage` is included. However, I could not figure out the correct way to do this. I believe you can use [`scryRenderedDOMComponentsWithTag`](https://facebook.github.io/react/docs/test-utils.html#scryrendereddomcomponentswithtag), but I could not get it to work. After searching for examples, it seems most people just default to checking if a particular class that is part of the component is included, which is what I eventually decided to use as well.